src/m365/spo/commands/tenant/tenant-settings-set.ts

```diff
@@ -97,6 +97,8 @@ export interface Options extends GlobalOptions {
   AllowedDomainListForSyncClient: string[];
   DisabledWebPartIds: string[];
   DisableCustomAppAuthentication: boolean;
+  EnableAzureADB2BIntegration: boolean;
```

Comment:
Since we're expecting people to specify a value these options should be a string, because people can type anything

Reply:
Well, I kind of aligned this with the existing props like DisableCustomAppAuthentication, which are also booleans. If you type anything other than 'true' or 'false' you get an error.

Reply:
Hmm, it could be that minimist is already converting string `true` to a boolean true for us. If true and false are coming across correctly then we can keep it as-is.

Reply:
For reference sake: there is some existing validation that forces you to write true or false, it's based on the autocomplete value:

```typescript
for (const item of commandOptions) {
  if (item.option.indexOf(propertyKey) > -1 &&
    item.autocomplete &&
    item.autocomplete.indexOf(propertyValue.toString()) === -1) {
    return `${propertyKey} option has invalid value of ${propertyValue}. Allowed values are ${JSON.stringify(item.autocomplete)}`;
  }
}
```

and this is what it looks like:
![image](https://user-images.githubusercontent.com/5267487/159232154-996804df-c820-4d68-800a-87e561fb7ebd.png)

I hadn't looked at minimist before, nice package! It's probably the reason the true converts to trueish indeed.

Reply:
Wait, I've changed my mind here.
The type is bool, but the value appears to be string if I debug. Which is the reason why I orignally put in a condition with a string value here: `if (args.options.EnableAzureADB2BIntegration === 'true') {}`

Reply:
I've got one more idea: we have the ability to force specific options to a string or boolean type. What if we used it here? Would it work as intended? See an example how we use it here: https://github.com/pnp/cli-microsoft365/blob/c51892964f48e7c11ad3a7365af991c072d3e191/src/m365/spo/commands/listitem/listitem-set.ts#L290-L292

Reply:
Ah, that must be the missing link with minimist. That works, thanks!